src/longbow/segment/command.py

```diff
@@ -594,21 +608,29 @@ def _write_split_array_element(
     if barcode_conf_file is not None and a.has_tag(bam_utils.READ_BARCODE_CONF_FACTOR_TAG):
         barcode_conf_file.write(f"{a.get_tag(bam_utils.READ_RAW_BARCODE_TAG)}\t{a.get_tag(bam_utils.READ_BARCODE_CONF_FACTOR_TAG)}\n")
 
-    bam_out.write(a)
+    has_cbc_and_umi = bam_utils.has_cbc_and_umi(a)
+
+    if not require_cbc_and_umi or has_cbc_and_umi:
+        bam_out.write(a)
 
 
 def create_simple_split_array_element(delim_name, end_coord, model, prev_delim_name, read, segments, start_coord):
     """Package an array element into an AlignedSegment from the results of simple splitting rules."""
-    a = pysam.AlignedSegment()
-    a.query_name = (
-        f"{read.query_name}/{start_coord}_{end_coord}/{prev_delim_name}-{delim_name}"
```

Comment:
Can you add another tag and preserve this information?  In the past it's been helpful for debugging.

Reply:
Done